app/components/safe_area_view/safe_area_view.ios.js

```diff
@@ -57,23 +58,52 @@ export default class SafeAreaIos extends PureComponent {
     componentDidMount() {
         this.mounted = true;
 
-        SafeArea.addEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
+        Dimensions.addEventListener('change', this.getSafeAreaInsets);
         EventEmitter.on('update_safe_area_view', this.getSafeAreaInsets);
+        if (EphemeralStore.safeAreaInsets.portait === null || EphemeralStore.safeAreaInsets.landscape === null) {
+            SafeArea.addEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
+        }
+
         this.keyboardDidShowListener = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
         this.keyboardDidHideListener = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
 
         this.getSafeAreaInsets();
     }
 
     componentWillUnmount() {
-        SafeArea.removeEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
+        Dimensions.removeEventListener('change', this.getSafeAreaInsets);
         EventEmitter.off('update_safe_area_view', this.getSafeAreaInsets);
+        SafeArea.removeEventListener('safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsForRootViewChange);
         this.keyboardDidShowListener.remove();
         this.keyboardDidHideListener.remove();
 
         this.mounted = false;
     }
 
+    getSafeAreaInsets = async (dimensions) => {
+        this.getStatusBarHeight();
+
+        if (DeviceTypes.IS_IPHONE_WITH_INSETS || mattermostManaged.hasSafeAreaInsets) {
+            const window = dimensions?.window || Dimensions.get('window');
+            const landscape = window.width > window.length;
+            const {safeAreaInsets} = await SafeArea.getSafeAreaInsetsForRootView();
```

Comment:
shouldn't we check if we already have the value on the ephemeral store and go for the safearea one if we don't?

Reply:
agree, as there can be various instances of the component and the values could be already in the Ephemeral store